### PR TITLE
Test Nuitka support for package Six

### PR DESCRIPTION
**Pull request by Pihu1998**
This is a test to check if PyPI package Six is supported by Nuitka

### CODE REVIEW

File: tests/standalone/six.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from __future__ import absolute_import
+import six
+import sys
+
+MAXSIZE = sys.maxsize
+
+def test_dispatch_types(value):
```
Comment (kayhayen) [inline tests/standalone/six.py]: The Nuitka test runner is executing these as programs, with CPython and Nuitka, and will check the outputs, and merely defining a function named `test_*` is not going to do anything to run it. You can put stuff in functions, but then you have to also call it. Most often for standalone tests, we do just put things on module level.

**Comment by Pihu1998** [comment]
@kayhayen  Please review the above changes and please inform if there is any room for further improvements.

File: tests/standalone/six.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from __future__ import absolute_import
+import six
+import sys
+
+MAXSIZE = sys.maxsize
+
+def dispatch_types(value):
```
Comment (kayhayen) [inline tests/standalone/six.py]: Well, not being named `test_*` still doesn't make the code being called, you realize that?

**Comment by Pihu1998** [comment]
@kayhayen Please review the above changes. I have included a main() function that calls all of the functions. Please inform if it can be improved further.

File: tests/standalone/six.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+PY3 = sys.version_info[0] == 3
+if PY3:
+ string_types = str,
+ integer_types = int,
+ class_types = type
+
+
+def dispatch_types(value):
+   
+    if isinstance(value, six.integer_types):
```
Comment (kayhayen) [inline tests/standalone/six.py]: This is not a big coverage of the six package, is it?

File: tests/standalone/six.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
+from __future__ import absolute_import
```
Comment (kayhayen) [inline tests/standalone/six.py]: You need that absolute_import only due to your strange naming of the test as "six", why not go with SixUsing.py to match what other tests have.

File: tests/standalone/six.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+def dispatch_types(value):
+   
+    if isinstance(value, six.integer_types):
+        print(value)
+
+    elif isinstance(value, six.string_types):
+        print(value)
+        
+    elif isinstance(value, six.class_types):
+        value
```
Comment (kayhayen) [inline tests/standalone/six.py]: That's not doing anything.

File: tests/standalone/six.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+ class_types = type
+
+
+def dispatch_types(value):
+   
+    if isinstance(value, six.integer_types):
+        print(value)
+
+    elif isinstance(value, six.string_types):
+        print(value)
```
Comment (kayhayen) [inline tests/standalone/six.py]: That's not distinguishable from above branch.

File: tests/standalone/six.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+        
+    elif isinstance(value, six.class_types):
+        value
+        
+
+def import_module(name):
```
Comment (kayhayen) [inline tests/standalone/six.py]: Why is this here?

File: tests/standalone/six.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    __import__(name)
+    return sys.modules[name]
+    
+
+def add_doc(func, doc):
```
Comment (kayhayen) [inline tests/standalone/six.py]: This is not using the doc parameter, what is it intended for?

**Comment by kayhayen** [comment]
@Pihu1998 Also do not just push new code, instead also respond to review comments I make, I failed to notice this one had changed, but assumed you didn't work on it.

File: tests/standalone/six.py (hunk 3 above)
Comment (Pihu1998) [reply tests/standalone/six.py]: Okay, I think I should expand the usage of Six specifically.

File: tests/standalone/six.py (hunk 4 above)
Comment (Pihu1998) [reply tests/standalone/six.py]: Yeah, Sure!

File: tests/standalone/six.py (hunk 7 above)
Comment (Pihu1998) [reply tests/standalone/six.py]: Oh sorry, my bad. I will remove it.

**Comment by kayhayen** [comment]
Well, maybe I was not precise enough. Respond to Github review comments right after you addressed them. Otherwise respond to ask a question about the comment or an outright refutation of it. But do not acknowledge, promise to make changes, or apologize.

I would much rather prefer to see a comment mean there is new code, or information I sought, so I can resolve the issue. Now there will be comments with responses, and I am left guessing when it makes sense to look at the code. That was not the point of asking you to reply, just to have something there, but rather to learn about what you have addressed.

**Comment by Pihu1998** [comment]
Sorry for the late push of code. I was working on it. I have added some more usage of the package Six in the code and also changed the test file name to 'SixUsing.py'. Please review the above changes.

File: tests/standalone/SixUsing.py [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    def test(self):
+        six.assertCountEqual(self, (1, 2), [2, 1])
+        
+
+def main():
+    print(dispatch_types(6))
+    print(dispatch_types('hi'))
+    u='hi'
+    print(dispatch_types(u.encode('utf-8')))
+    Dict = {'Tim': 18,'Charlie':12,'Tiffany':22,'Robert':25}
```
Comment (kayhayen) [inline tests/standalone/SixUsing.py]: Just a minor quibble, don't name local variables this way, makes me think it's a class or when I see it used.

As for the prints, import the `print_function` from future, and add some identifying prefixes, e.g. `dispatch_types int:`, the so output is more human readable. It works the way you did it, but it's not nice.

File: tests/standalone/six.py (hunk 4 above)
Comment (kayhayen) [reply tests/standalone/six.py]: Won't you delete this file then, it seems you forgot about that?

File: tests/standalone/SixUsing.py [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        return str
+    
+    elif isinstance(value, six.binary_type):
+        return bytes
+
+def get_iterator(dictionary):
+    return six.viewkeys(dictionary), six.viewvalues(dictionary), six.viewitems(dictionary)
+
+
+class TestAssertCountEqual(unittest.TestCase):
```
Comment (kayhayen) [inline tests/standalone/SixUsing.py]: So, is this is a six extension for unit tests here?

**Comment by kayhayen** [comment]
Unlikely to be finished.
